hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/TokenAccessorImpl.java

```diff
@@ -0,0 +1,173 @@
+    public Optional<EvmTokenInfo> evmInfoForToken(Address token, ByteString ledgerId) {
+        return trackingLedgers.evmInfoForToken(tokenIdFromEvmAddress(token), ledgerId);
+    }
+
+    @Override
+    public Optional<EvmNftInfo> evmNftInfo(
+            final Address token, long serialNo, final ByteString ledgerId) {
+        final var target =
+                NftID.newBuilder()
+                        .setSerialNumber(serialNo)
+                        .setTokenID(tokenIdFromEvmAddress(token))
+                        .build();
+        return trackingLedgers.evmNftInfo(target, ledgerId);
+    }
+
+    @Override
+    public boolean isTokenAddress(Address address) {
+        return trackingLedgers.isTokenAddress(address);
+    }
+
+    @Override
+    public boolean isFrozen(Address account, Address token) {
+        return trackingLedgers.isFrozen(
+                accountIdFromEvmAddress(account), tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public boolean defaultFreezeStatus(Address token) {
+        return trackingLedgers.defaultFreezeStatus(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public boolean defaultKycStatus(Address token) {
+        return trackingLedgers.defaultKycStatus(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public boolean isKyc(Address account, Address token) {
+        return trackingLedgers.isKyc(
+                accountIdFromEvmAddress(account), tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public Optional<List<CustomFee>> infoForTokenCustomFees(Address token) {
+        return trackingLedgers.infoForTokenCustomFees(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public TokenType typeOf(Address token) {
+        return trackingLedgers.typeOf(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public EvmKey keyOf(Address token, TokenKeyType keyType) {
+        final var key =
+                trackingLedgers.keyOf(
+                        tokenIdFromEvmAddress(token), TokenProperty.valueOf(keyType.name()));
+
+        return convertToEvmKey(asKeyUnchecked(key));
+    }
+
+    @Override
+    public String nameOf(Address token) {
+        return trackingLedgers.nameOf(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public String symbolOf(Address token) {
+        return trackingLedgers.symbolOf(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public long totalSupplyOf(Address token) {
+        return trackingLedgers.totalSupplyOf(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public int decimalsOf(Address token) {
+        return trackingLedgers.decimalsOf(tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public long balanceOf(Address account, Address token) {
+        return trackingLedgers.balanceOf(
+                accountIdFromEvmAddress(account), tokenIdFromEvmAddress(token));
+    }
+
+    @Override
+    public long staticAllowanceOf(Address owner, Address spender, Address token) {
```

Comment:
The static context has more meaning for the `trackingLedgers` internal logic. I believe on this `TokenAccessor` level we can skip it. So we can rename the method something like `staticAllowanceOf` -> `allowanceOf`